### ISSUE TITLE
ClaimTypes.Sid has a different behaviour than the other ClaimTypes

### ISSUE THREAD

**Issue by ranouf** [dotnet/corefx#28454]
Hi,

I m using the namespace System.Security.Claims to generate a new JWT 

```
 var tokenDescriptor = new SecurityTokenDescriptor
            {
                Issuer = authSettings.Issuer,
                Audience = authSettings.Audience,
                NotBefore = DateTime.UtcNow,
                IssuedAt = DateTime.UtcNow,
                Subject = new ClaimsIdentity(new Claim[]
                {
                    new Claim(ClaimTypes.Sid, user.Id.ToString()),
                    new Claim(ClaimTypes.GivenName, user.Firstname),
                    new Claim(ClaimTypes.Surname, user.Lastname),
                    new Claim(ClaimTypes.Email, user.Email),
                    new Claim(ClaimTypes.Role, role.Name),
                }),
                Expires = token.ExpirationDate,
                SigningCredentials = GenerateSigningCredentials(authSettings.SecretKey)
            };
```

When I decode the JWT (for exemple on https://jwt.io/) I can see:

```
{
  **"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/sid"**: "37cf7320-5669-4dbb-767f-08d58ddd8d0b",
  "given_name": "XX",
  "family_name": "XX",
  "email": "XX@XX.com",
  "role": "XX",
  "nbf": 1521957359,
  "exp": 1522562150,
  "iat": 1521957359,
  "iss": "XX",
  "aud": "http://localhost:9334/"
}

```

Is there a specific reason to encode *http://schemas.xmlsoap.org/ws/2005/05/identity/claims/sid* in the JWT instead of *sid*?
In my opinion, the behaviour must the same for all the ClaimsType.

Can you fix that?

**Comment by bartonjs** [comment]
Is that `Microsoft.IdentityModel.Tokens.SecurityTokenDescriptor`?  If so, that belongs at https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet.  But a quick check turned up https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/421, which is theoretically fixed in the current version (5.2.1)

**Comment by ranouf** [comment]
Thanks @bartonjs , this is a perfect answer!
